controllers/vmware/vspherecluster_reconciler.go

```diff
@@ -369,9 +370,68 @@ func (r *ClusterReconciler) VSphereMachineToCluster(ctx context.Context, o clien
 	}}
 }
 
+// ZoneToVSphereClusters adds reconcile requests for VSphereClusters when Zone has an event.
+func (r *ClusterReconciler) ZoneToVSphereClusters(ctx context.Context, o client.Object) []reconcile.Request {
+	log := ctrl.LoggerFrom(ctx)
+
+	zone, ok := o.(*topologyv1.Zone)
+	if !ok {
+		log.Error(nil, fmt.Sprintf("Expected a Zone but got a %T", o))
+		return nil
+	}
+	log = log.WithValues("Zone", klog.KObj(zone))
+	ctx = ctrl.LoggerInto(ctx, log)
```

Comment:
```suggestion
```

No need to pass this in, won't be used.

Reply:
Only if we assume that client-go doesn't use the logger from context. Might be or might not be true at the moment, might become true in the future.

In general I prefer always updating the ctx, just in case (also nobody will remember updaing this if we oureslves start using the logger from ctx a bit further down in the future)